### PR TITLE
feat: Add product flavor for fdroid

### LINKED ISSUE #1240: Add product flavour for fdroid
### Feature Request

**Feature Description**
To launch open-event-orga-app on fdroid, product flavours need to be added as the first step.

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by sridharjajoo**
Fixes #1240 

Checklist:

- [ ] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -65,6 +65,21 @@ android {
...
+    productFlavors {
+        fdroid {
+            dimension "default"
+            applicationIdSuffix ".fdroid"
+            versionNameSuffix "-FDroid"
+        }
+
+        playStore {
+            dimension "default"
+            versionNameSuffix "-PlayStore"
```
Comment (iamareebjamal) [inline app/build.gradle]: Don't change version name

File: app/build.gradle [hunk 2]
```diff
@@ -65,6 +65,20 @@ android {
...
         }
     }
+
+    flavorDimensions "default"
+
+    productFlavors {
+        fdroid {
+            dimension "default"
+            applicationIdSuffix ".fdroid"
+            versionNameSuffix "-FDroid"
```
Comment (iamareebjamal) [inline app/build.gradle]: Same here

**Comment by sridharjajoo** [comment]
@iamareebjamal I have done the changes. But build is failing and the message is
 ```Task 'testDebugUnitTestCoverage' not found in root project 'open-event-orga-app'.```

**Comment by iamareebjamal** [comment]
Yes now it will have testFdroidDebug...

**Comment by sridharjajoo** [comment]
@Masquerade0097 Will need to change the test name in circle ci as well. I'm updating it.

**Comment by sridharjajoo** [comment]
@iamareebjamal Ok I'll do the required changes.
Also with regards to the google play services and other proprietary libraries which fdroid doesn't allow, we are using 3 play services libraries : Places, Vision and Open source software. So what other libraries do you suggest as an alternative. For vision, I am thinking [link](https://github.com/zxing/zxing) , for open source software we might hide it or use [link](https://github.com/PSDev/LicensesDialog)

**Comment by iamareebjamal** [comment]
Both seem great

**Comment by sridharjajoo** [comment]
@iamareebjamal Done the changes. Please review.

File: scripts/update-apk.sh [hunk 3]
```diff
@@ -18,12 +18,15 @@ cd apk
 if [ "$TRAVIS_BRANCH" == "$PUBLISH_BRANCH" ]; then
 	/bin/rm -f *
 else
-	/bin/rm -f app-debug.apk app-release.apk
+	/bin/rm -f app-debug.apk app-release.apk app-fdroid-debug.apk app-fdroid-release.apk
```
Comment (nikit19) [inline scripts/update-apk.sh]: This should be `app-playStore-debug.apk` right? same for release

Comment (iamareebjamal) [reply scripts/update-apk.sh]: Yes

**Comment by sridharjajoo** [comment]
@iamareebjamal Made the changes. Please review

**Pull request merged**